CMake/modules/PrecompiledHeader.cmake

```diff
@@ -0,0 +1,249 @@
+  set(_compile_options "$<$<BOOL:${_compile_options}>:$<JOIN:${_compile_options},\n>\n>")
+  set(_define_symbol "$<$<BOOL:${_define_symbol}>:-D$<JOIN:${_define_symbol},\n-D>\n>")
+  file(GENERATE OUTPUT "${_filename}" CONTENT "${_compile_definitions}${_include_directories}${_compile_flags}${_compile_options}${_define_symbol}\n")
+endfunction()
+
+macro(add_precompiled_header_cpp _target)
+  if(MSVC)
+    list(APPEND ${FWPROJECT_NAME}_SOURCES
+        "${FWCMAKE_RESOURCE_PATH}/build/pch.cpp"
+        "${CMAKE_CURRENT_SOURCE_DIR}/include/${_target}/pch.hpp"
+    )
+  endif()
+endmacro()
+
+function(add_precompiled_header _target _input)
+  cmake_parse_arguments(_PCH "FORCEINCLUDE" "SOURCE_CXX:SOURCE_C" "" ${ARGN})
+
+  get_filename_component(_input_we ${_input} NAME_WE)
+  get_filename_component(_input_pch ${_input} NAME)
+
+  set(_PCH_SOURCE_CXX "${FWCMAKE_RESOURCE_PATH}/build/pch.cpp")
+
+  if(MSVC)
+
+    set(_cxx_path "${CMAKE_CURRENT_BINARY_DIR}/cxx_pch")
+    make_directory("${_cxx_path}")
+    set(_pch_cxx_header "${CMAKE_CURRENT_SOURCE_DIR}/include/${_target}/pch.hpp")
+    set(_pch_cxx_pch "${_cxx_path}/${_input_we}.pch")
+
+    get_target_property(sources ${_target} SOURCES)
+    foreach(_source ${sources})
+      set(_pch_compile_flags "")
+      if(_source MATCHES \\.\(cc|cxx|cpp\)$)
+
+        if(_source STREQUAL "${_PCH_SOURCE_CXX}")
+          set(_pch_compile_flags "${_pch_compile_flags} \"/Fp${_pch_cxx_pch}\" /Yc${_input_pch} \"/I${CMAKE_CURRENT_SOURCE_DIR}\\include\\${_target}\"")
+          set(_pch_source_cxx_found TRUE)
+          set_source_files_properties("${_source}" PROPERTIES OBJECT_OUTPUTS "${_pch_cxx_pch}")
+        else()
+          set(_pch_compile_flags "${_pch_compile_flags} \"/Fp${_pch_cxx_pch}\" /Yu${_input}")
+          set(_pch_source_cxx_needed TRUE)
+          set_source_files_properties("${_source}" PROPERTIES OBJECT_DEPENDS "${_pch_cxx_pch}")
+          # Force the include of the PCH on every source file
+          set(_pch_compile_flags "${_pch_compile_flags} /FI${_input}")
+        endif()
+
+        get_source_file_property(_object_depends "${_source}" OBJECT_DEPENDS)
+        if(NOT _object_depends)
+          set(_object_depends)
+        endif()
+
+        list(APPEND _object_depends "${_pch_cxx_header}")
+
+        set_source_files_properties(${_source} PROPERTIES
+          COMPILE_FLAGS "${_pch_compile_flags}"
+          OBJECT_DEPENDS "${_object_depends}")
+      endif()
+    endforeach()
+
+    list(APPEND CMAKE_PCH_COMPILER_TARGETS ${_target})
+    set(CMAKE_PCH_COMPILER_TARGETS "${CMAKE_PCH_COMPILER_TARGETS}" PARENT_SCOPE)
+
+    if(_pch_source_cxx_needed AND NOT _pch_source_cxx_found)
+      message(FATAL_ERROR "A source file ${_PCH_SOURCE_CXX} for ${_input} is required for MSVC builds. Can be set with the SOURCE_CXX option.")
+    endif()
+  endif(MSVC)
+
+  if(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
```

Comment:
On MacOS, the CMAKE_CXX_COMPILER_ID is "AppleClang". 
We should use MATCHES instead of STREQUAL

It works using:
   `if(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")`